### PR TITLE
Reformatting, switch to Mamba

### PR DESCRIPTION
**Pull request by mvansluijs**

### CODE REVIEW

**Comment by mvansluijs** [comment]
@GreatEmerald

File: index.Rmd [hunk 1]
```diff
@@ -1,69 +1,73 @@
...
 
-### Conda installation
-Conda can be downloaded in two versions: Anaconda and Miniconda. The difference is that Miniconda is just shipping the repository management system. So when it is installed, you will only have the management system without packages. Whereas with Anaconda, you get the distribution with most popular Python packages already built in.
+Mamba replaces Conda for most commands, *except for activating and deactivating environments*. You will see more clearly how this works in practice in the first part of this tutorial.
 
-In Geo-scripting Miniconda is used. To install Miniconda in your Linux environment, we have prepared a short _Bash_ script for you. Just run the following lines of code, line by line, in a new terminal window.
+To install Mamba in your Linux environment, we have prepared a short `Bash` script for you. Just run the following lines of code, line by line, in a new terminal window.
```
Comment (GreatEmerald) [inline index.Rmd]: `Bash` is not a command, though. It's a name of the language, hence why it's in italics.

File: index.Rmd [hunk 2]
```diff
@@ -1,69 +1,73 @@
...
+## Python package management with Conda
+A set of tools co-exist for installing and managing Python packages. It is possible to install packages on your base Python interpreter, but sooner or later you will get conflicting Python packages since packages have varying dependencies. It can even [break your system Python interpreter](https://askubuntu.com/questions/95037/what-is-the-best-way-to-install-python-packages).
 
+Instead, we recommend to create a `conda` environment on your local machine for each project. In `conda` environments, basically anything, such as software, C libraries or R packages can be installed. Here we use them here for installing Python packages. Packages installed in one environment do not interfere with your base Python or with other `conda` environments.
```
Comment (GreatEmerald) [inline index.Rmd]:
```suggestion
Instead, we recommend to use a Python package manager that can make use of virtual environments, such as Conda or Mamba. That way, you can create a `conda` environment on your local machine for each project. In `conda` environments, basically anything, such as software, C libraries or R packages can be installed. Here we use them here for installing Python packages. Packages installed in one environment do not interfere with your base Python or with other `conda` environments.
```

File: index.Rmd [hunk 3]
```diff
@@ -89,107 +93,101 @@ Scroll down the license with enter. Accept License (i.e. type yes). Use default
...
 
-This will install Miniconda into `~/miniconda3`. Finally, restart your terminal to be able to use `conda` in the terminal. Next, let's see how to use Conda in case you want make new virtual environments by yourself, or install packages after creating the environment.
+This will install Mamba into `~/mamba`. Finally, restart your terminal to be able to use `mamba` and `conda` in the terminal. Next, let's see how to use Mamba in case you want make new virtual environments by yourself, or install packages after creating the environment.
 
-### Conda usage
-Conda creates isolated environments with sets of packages, that do not interfere with your base Python or with other conda environments. To create an environment:
+## Mamba usage
+Mamba creates isolated `conda` environments with sets of packages, that do not interfere with your base Python or with other `conda` environments. To create an environment:
```
Comment (GreatEmerald) [inline index.Rmd]: Same goes here, `conda` is a command, but in this case we are not talking about the command, but about the *Conda* environments.

File: index.Rmd [hunk 4]
```diff
@@ -89,107 +93,101 @@ Scroll down the license with enter. Accept License (i.e. type yes). Use default
...
 
 ```{r, eval=FALSE, engine='bash'}
-conda create --name geotest python=2.7 numpy
+mamba create --name geotest python numpy
 ```
 
-This would create a new environment called _geotest_ with Python 2.7 and _NumPy_ installed into the conda environment. Another option is to create an environment from a `.yaml` file, in which all required modules are listed; see the example later on. To create an environment from such a file, use the argument `--file` (or `--f` in short), see the conda (documentation)[https://docs.conda.io/projects/conda/en/latest/commands/create.html]. To list the available environments:
+This would create a new environment called `geotest` with Python 2.7 and `NumPy` and `Spyder` installed into the `conda` environment. Another option is to create an environment from a `.yaml` file, in which all required modules are listed. You will see an example of this later on in this tutorial. To create an environment from such a file, you can use the argument `--file` (or `--f` in short).
```
Comment (GreatEmerald) [inline index.Rmd]: Also here, it's called *geotest* but it's not a command called `geotest`. *NumPy* and *Spyder* are also not commands.

File: index.Rmd [hunk 5]
```diff
@@ -89,107 +93,101 @@ Scroll down the license with enter. Accept License (i.e. type yes). Use default
...
+After this, the current environment is shown in parentheses in front of your prompt (`(geotest)$`). Note that the activated environment is only valid for the shell in which you activated it. For instance, if you close the shell window and open a new one you will have to activate it again.
 
-After creating a conda environment, (additional) Python packages can be installed. There are three recommended ways to install packages, which we list below. In the documentation of most Python packages the recommended way to install them can be found.
+After creating a `conda` environment, (additional) Python packages can be installed. There are three possible ways to install packages, which we list below. In the documentation of most Python packages the recommended way to install them can be found.
```
Comment (GreatEmerald) [inline index.Rmd]: I would leave out the part about looking at package documentation. The documentation almost every time says to use pip, but that does not mean that it's a good idea to do that in a conda environment.

File: index.Rmd [hunk 6]
```diff
@@ -89,107 +93,101 @@ Scroll down the license with enter. Accept License (i.e. type yes). Use default
...
 
-* Using [conda](https://docs.conda.io/projects/conda/en/latest/index.html)  to install and manage packages. Conda packages are downloaded from conda channels, which are URLs to directories containing conda packages. Generally, installing with conda is the preferred method.
-* Using _pip_ to install packages and _conda_ to manage packages. _Pip_ is available for Windows, macOS and Linux. Pip can also install [binary wheels on Windows](https://www.lfd.uci.edu/~gohlke/Pythonlibs).
+* Using `mamba` to install and manage `conda` packages. This downloads `conda` packages using `conda` channels, which are URLs to directories containing the `conda` packages. **Generally, installing `conda` packages using `mamba` is the preferred method.**
+* Using `pip` to install packages and `mamba` to manage these packages. `pip` is available for Windows, macOS and Linux. `pip` can also install [binary wheels on Windows](https://www.lfd.uci.edu/~gohlke/Pythonlibs).
```
Comment (GreatEmerald) [inline index.Rmd]: It would be handy here to state that you should generally not install packages from `pip` in a conda environment unless it's the last resort. This is because after you use `pip` to modify an environment, you can no longer use `conda`/`mamba` to do so (trying that will break your environment, because pip does not communicate its changes to conda). Hence install packages with mamba that you can first, and only then use pip, and then never touch the environment with mamba again (delete and start fresh if you need to).

File: index.Rmd [hunk 7]
```diff
@@ -1,69 +1,73 @@
...
 
+## Conda/Mamba installation
+For this course, we will make use of Mamba, fast drop-in reimplementation of the Conda package manager. It has its core parts implemented in C++ for maximum efficiency, makes use of parallel downloading of repository data and package files using multi-threading, and uses `libsolv` for (much) faster dependency solving. 
 
-### Conda installation
-Conda can be downloaded in two versions: Anaconda and Miniconda. The difference is that Miniconda is just shipping the repository management system. So when it is installed, you will only have the management system without packages. Whereas with Anaconda, you get the distribution with most popular Python packages already built in.
+Mamba replaces Conda for most commands, *except for activating and deactivating environments*. You will see more clearly how this works in practice in the first part of this tutorial.
```
Comment (GreatEmerald) [inline index.Rmd]: But we want students to use `source` for activating and deactivating, so this is probably not so useful to mention anyway.

File: index.Rmd [hunk 8]
```diff
@@ -89,107 +93,101 @@ Scroll down the license with enter. Accept License (i.e. type yes). Use default
...
 ```{r, eval=FALSE, engine='bash'}
-conda info --envs
+mamba info --envs
 ```
 
-Conda puts an asterisk (*) in front of the active environment. To activate the environment:
+Mamba puts an asterisk (*) in front of the active environment. To activate the environment:
 
 ```{r, eval=FALSE, engine='bash'}
-## Cross-platform (but not always working, like in our VDI, try it)
```
Comment (GreatEmerald) [inline index.Rmd:110]: It indeed does not work by default. It would be good to mention this in some way. If you try it, it will complain that `conda init` was not run, but doing so does fairly invasive changes to the `~/.bashrc` file that are not needed, because `source` works fine.

File: index.Rmd [hunk 9]
```diff
@@ -696,21 +664,20 @@ for element in compound:
...
-
-# LOOPING A LIST
+# Looping a list
 campus = ['Gaia', 'Lumen', 'Radix', 'Forum']
 for i in campus:
     print(i)
 
-# LOOPING A STRING
-our_building = campus[0] # GAIA
+# Looping
```
Comment (GreatEmerald) [inline index.Rmd]: ...over a string?

**Review by GreatEmerald — CHANGES_REQUESTED** [review]
Thanks, some suggestions.

File: index.Rmd (hunk 7 above)
Comment (mvansluijs) [reply index.Rmd]: Intuitively, I would personally just think it would be `mamba activate` or something like that. I feel it makes sense to mention that it does not work this way (or at least by default it doesn't), even if we later opt for using `source`. It is well explained there how&why we do not use `conda activate`. I'm in favor of leaving this part in.

File: index.Rmd [hunk 10]
```diff
@@ -88,32 +88,32 @@ popd
...
 ```{r, eval=FALSE, engine='bash'}
 mamba info --envs
 ```
 
-Mamba puts an asterisk (*) in front of the active environment. To activate the environment:
+*Mamba* puts an asterisk (*) in front of the active environment. To activate the environment:
 
 ```{r, eval=FALSE, engine='bash'}
-# Cross-platform (but not always working)
+# Cross-platform (but not always working, like in our VDI, so we use the next option)
```
Comment (GreatEmerald) [inline index.Rmd]: Better to use the term "VM", because VDI is actually only the VMWare Horizon VDIs, and not the SURF Research Cloud, whereas "VM" means both.

**Review by GreatEmerald — CHANGES_REQUESTED** [review]
One more small thing:

File: index.Rmd (hunk 7 above)
Comment (GreatEmerald) [reply index.Rmd]: Well, that's OK, but maybe rephrase it, because right now it seems like a very important piece of information (that in the end turns out actually useless in practice).

Comment (mvansluijs) [reply index.Rmd]: I've moved it to the section where it is relevant and rephrased it. From context, it should be clearer now.

Comment (GreatEmerald) [reply index.Rmd]: OK!

**Review by GreatEmerald — APPROVED** [review]
Thanks!
